x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/components/agent_upgrade_status.tsx

```diff
@@ -103,6 +107,28 @@ function getStatusComponents(agentUpgradeDetails?: AgentUpgradeDetails) {
         ),
       };
     case 'UPG_DOWNLOADING':
+      if (agentUpgradeDetails?.metadata?.retry_error_msg) {
+        return {
+          Badge: (
+            <EuiBadge color="accent" iconType="download">
+              <FormattedMessage
+                id="xpack.fleet.agentUpgradeStatusBadge.upgradeDownloading"
+                defaultMessage="Upgrade downloading"
+              />
+            </EuiBadge>
+          ),
+          WarningTooltipText: (
+            <FormattedMessage
+              id="xpack.fleet.agentUpgradeStatusTooltip.upgradeDownloadingFailed"
+              defaultMessage="Upgrade failing: {retryMsg}. {retryUntil}"
+              values={{
+                retryMsg: agentUpgradeDetails?.metadata?.retry_error_msg,
+                retryUntil: formatRetryUntil(agentUpgradeDetails?.metadata?.retry_until),
```

Comment:
Since the retry_until is a deadline in UTC, it might be easier for the user to read to if we also calculated and presented the time remaining as `retry_until - now()`. This would let us present something like "Retrying until: 2024-01-30T15:03:38.159Z (1h53m remaining)" to save the user from having to calculate this in their own head.

Reply:
I added the remaining time in humanized format as discussed:
![Screenshot 2024-01-02 at 16 54 28](https://github.com/elastic/kibana/assets/16084106/32677f95-b20f-49f8-9bff-e17ae1d73e25)

I'm not displaying it at all in those cases when the retry time is set in the past and the difference is negative. It can happen when the upgrade failed and value doesn't change anymore.